checkov/terraform/runner.py

```diff
@@ -65,6 +65,7 @@ def __init__(
         self.graph_registry = get_graph_checks_registry(self.check_type)
         self.definitions_with_modules: Dict[str, Dict] = {}
         self.referrer_cache: Dict[str, str] = {}
+        self.non_referred_cache: List[str] = []
```

Comment:
```suggestion
        self.non_referred_cache: Set[str] = set()
```
of course `Set` needs to be imported